src/index.js

```diff
@@ -1,282 +1,658 @@
+      if (originalResponse.readyState === 4) {
+        const jsonAttestation = JSON.parse(originalResponse.responseText)
+        attestationPdf.push(jsonAttestation)
+      }
+    })
+    return openProxied.apply(this, [].slice.call(arguments))
+  }
+  if (arguments[1].includes('/mesremboursements/?debutPeriode')) {
+    originalResponse.addEventListener('readystatechange', function () {
+      if (originalResponse.readyState === 4) {
+        const jsonBills = JSON.parse(originalResponse.responseText)
+        foundBills.push(jsonBills)
+      }
+    })
+    return openProxied.apply(this, [].slice.call(arguments))
+  }
+  // eslint-disable-next-line no-useless-escape
+  if (arguments[1].match(/\/mesremboursements\/\d+/g)) {
+    originalResponse.addEventListener('readystatechange', function () {
+      if (originalResponse.readyState === 4) {
+        const jsonDetails = JSON.parse(originalResponse.responseText)
+        foundBillsDetails.push(jsonDetails)
+      }
+    })
+    return openProxied.apply(this, [].slice.call(arguments))
+  }
+  if (arguments[1].includes('/mesremboursements/edition?debutPeriode=')) {
+    originalResponse.addEventListener('readystatechange', function () {
+      if (originalResponse.readyState === 4) {
+        const base64 = JSON.parse(originalResponse.responseText)
+        base64Pdfs.push(base64.edition)
+      }
+    })
+    return openProxied.apply(this, [].slice.call(arguments))
+  }
+  return openProxied.apply(this, [].slice.call(arguments))
+}
 
-  log('info', 'Fetching the list of documents')
-  const $ = await request(
-    `${baseUrl}/EspaceAdherentWebApp/MesDecomptes/Accueil`
-  )
+class LaMutuelleGeneraleContentScript extends ContentScript {
+  onWorkerReady() {
+    this.log('info', 'onWorkerReady starts')
+    const form = document.querySelector('form')
+    const loginError = document.querySelector('span[id*="error-element-"]')
+    if (form) {
+      form.addEventListener('submit', () => {
+        this.log('info', 'Form submit detected, sending credentials')
+        const password = document.querySelector('#password')?.value
+        const login = document.querySelector('#username')?.value
+        this.bridge.emit('workerEvent', {
+          event: 'loginSubmit',
+          payload: { login, password }
+        })
+      })
+    }
+    if (loginError) {
+      this.bridge.emit('workerEvent', {
+        event: 'loginError',
+        payload: { msg: loginError.innerHTML }
+      })
+    }
+  }
 
-  log('info', 'Parsing list of bills')
-  const documents = await parseBills($)
+  async onWorkerEvent({ event, payload }) {
+    this.log('info', 'onWorkerEvent starts')
+    if (event === 'loginSubmit') {
+      this.log('info', 'received loginSubmit, blocking user interactions')
+      this.blockWorkerInteractions()
+      const { login, password } = payload || {}
+      if (login && password) {
+        // On this website you could use your adherent number or your mail.
```

Comment:
And what happens if the user does login with  the adherent number ?

Reply:
Nothing special, it's the case with the account we have to develop, I just wanted to nit this in case someone were wondering why we save "email" when they give "number". As I said in the comment it's just to respect the "convention" and to not have to think about wether we should use "email" or "login" (or else) when manipulating the credentials 👍